python/feast_spark/pyspark/historical_feature_retrieval_job.py

```diff
@@ -591,10 +591,17 @@ def _read_and_verify_feature_table_df_from_source(
     feature_table_dtypes = dict(mapped_source_df.dtypes)
     for field in feature_table.entities + feature_table.features:
         column_type = feature_table_dtypes.get(field.name)
+
         if column_type != field.spark_type:
-            raise SchemaError(
-                f"{field.name} should be of {field.spark_type} type, but is {column_type} instead"
-            )
+            # For batch sources that does not have a float type, such as BigQuery
+            if column_type == "double" and field.spark_type == "float":
```

Comment:
can we generalize that a little? smth like constant `ALLOWED_CAST = [("double", "float"), ("float", "double")]`?
```
if (column_type, filed.spark_type) in ALLOWED_CAST:
...
```